oioioi/contests/utils.py

```diff
@@ -247,6 +256,96 @@ def visible_rounds(request):
     return [r for r in queryset if controller.can_see_round(request, r)]
 
 
+@request_cached
+def get_number_of_rounds(request):
+    """Returns the number of rounds in the current contest.
+    """ 
+    return Round.objects.filter(contest=request.contest).count()
```

Comment:
this can be replaced with `get_visible_rounds` if I am correct.

Reply:
visible_rounds() returns only rounds that a user can see, and by deafult it does not show the rounds that will start in the future. My train of thought was to inform the user how many rounds are in the contest including ones in the future. However, I may change it if counting only visible rounds is a better option (as seeing that contest has more rounds than the visible ones might be confusing for some users).

Reply:
Okay, that's a problem where either solution could be chosen, because one or the other can be better depending on a situation. I am not in a position to make a final decision here, but I'd say all rounds should be counted, because it doesn't really reveal any crucial information about contest's problems.

Reply:
Also, since hiding the rules will be made possible (or already has been), then it is always possible to hide this information if needed.